include/Standalone/Dialect/Check/CheckOps.td

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CHECK_DIALECT_CHECK_OPS
+#define CHECK_DIALECT_CHECK_OPS
+
+include "mlir/IR/OpBase.td"
+
+def CHECK_Dialect : Dialect {
+  let name = "check";
+  let cppNamespace = "::mlir::check";
+  let summary = [{
+    A dialect implementing test assertions.
+  }];
+}
+
+def CHECK_ExpectTrueOp : Op<CHECK_Dialect, "expect_true"> {
+  let summary = [{Checks that the operand is true}];
+  let description = [{
+    Verifies that the operand contains a true value, which is represented by
+    any non-zero integer.
+
+    Issues a non-fatal failure if the verification fails.
+
+    ```mlir
+    check.expect_true(%arg0) : i32
+    ```
+  }];
+
+  let arguments = (ins AnySignlessInteger:$operand);
+
+  let assemblyFormat = "`(` $operand `)` attr-dict `:` type($operand)";
+}
+
+def CHECK_ExpectAlmostEqOp :
+    Op<CHECK_Dialect, "expect_almost_eq", [SameTypeOperands]> {
+  let summary = [{Checks that the operands are almost equal}];
+  let description = [{
+    Verifies that the tensor operands with float elements are
+    almost equal to within an implementation-defined "reasonable" tolerance.
```

Comment:
I think what's _reasonable_ should be a part of this op, via an attribute. We can start with a simple fp-comparison and a delta (which is the attribute), but we could upgrade this into more complex attributes to handle statistical analysis later.

Reply:
Agree. I think TOSA has something along this line.